File: python/square/MathOperationsClient.py
import sys, glob
sys.path.append('.')
sys.path.insert(0, glob.glob('/home/tog/programs/thrift/lib/py/build/lib.*')[0])

import MathOperations
from ttypes import *

from thrift import Thrift
from thrift.transport import TSocket
from thrift.transport import TTransport
from thrift.protocol import TBinaryProtocol


if len(sys.argv) != 3:
	print("Usage: MathOperationsClient <server_port> <number>")
	sys.exit(1);

transport = TSocket.TSocket('localhost', int(sys.argv[1]))

protocol = TBinaryProtocol.TBinaryProtocol(transport)

client = MathOperations.Client(protocol)

transport.open()

print("Result: {}".format(client.square(int(sys.argv[2]))));

File: java/square/src/main/java/com/math/MathOperationsClient.java
package main.java.com.math;

import org.apache.thrift.TException;
import org.apache.thrift.protocol.TBinaryProtocol;
import org.apache.thrift.protocol.TProtocol;
import org.apache.thrift.transport.TSocket;
import org.apache.thrift.transport.TTransport;

public class MathOperationsClient {
	
	static TTransport transport;
	
	public static void main(String args[]){
		
		if(args.length!=2){
			System.out.println("Usage: MathOperationsClient <server port> <number>");
			System.exit(1);
		}
		
		transport = new TSocket("localhost", Integer.valueOf(args[0]));
		
		try{
			transport.open();
			
			TProtocol protocol = new  TBinaryProtocol(transport);
			MathOperations.Client client = new MathOperations.Client(protocol);
			
			System.out.println("Result:"+client.square(Integer.valueOf(args[1])));
		
		} catch (TException x) {
			x.printStackTrace();
	    } 
	}
}
